app/tests/spec/lib/fxa-client.js

```diff
@@ -646,6 +646,180 @@ define(function (require, exports, module) {
       });
     });
 
+    describe('sessionReauth', function () {
+      var sessionToken = 'session token';
+
+      it('reauth with invalid session should faill', function () {
+        sinon.stub(realClient, 'sessionReauth').callsFake(function () {
+          return Promise.reject(AuthErrors.toError('INVALID_TOKEN'));
+        });
+
+        return client.sessionReauth('badSessionToken', email, password, relier)
+          .then(assert.fail, function (err) {
+            assert.isTrue(AuthErrors.is(err, 'INVALID_TOKEN'));
+          });
+      });
+
+      it('reauth w/ relier that wants keys submits email/password and returns keys', function () {
+        sinon.stub(realClient, 'sessionReauth').callsFake(function () {
+          return Promise.resolve({
+            keyFetchToken: 'keyFetchToken',
+            unwrapBKey: 'unwrapBKey',
+            verificationMethod: VerificationMethods.EMAIL,
+            verificationReason: VerificationReasons.SIGN_IN,
+            verified: false
+          });
+        });
+
+        sinon.stub(relier, 'wantsKeys').callsFake(() => true);
+
+        return client.sessionReauth(sessionToken, email, password, relier, { resume: resumeToken })
+          .then(function (sessionData) {
+            assert.isTrue(realClient.sessionReauth.calledWith(sessionToken, trim(email), password, {
+              keys: true,
+              reason: SignInReasons.SIGN_IN,
+              redirectTo: REDIRECT_TO,
+              resume: resumeToken,
+              service: SYNC_SERVICE
+            }));
+
+            assert.equal(sessionData.keyFetchToken, 'keyFetchToken');
+            assert.equal(sessionData.unwrapBKey, 'unwrapBKey');
+            assert.isFalse(sessionData.verified);
+            assert.equal(sessionData.verificationMethod, VerificationMethods.EMAIL);
+            assert.equal(sessionData.verificationReason, VerificationReasons.SIGN_IN);
+          });
+      });
+
+      it('reauth w/ relier that does not want keys submits email/password and does not request keys', function () {
+        sinon.stub(realClient, 'sessionReauth').callsFake(() => Promise.resolve({}));
+
+        relier.set('service', NON_SYNC_SERVICE);
+        sinon.stub(relier, 'wantsKeys').callsFake(() => false);
+
+        return client.sessionReauth(sessionToken, email, password, relier)
+          .then(function (sessionData) {
+            assert.isTrue(realClient.sessionReauth.calledWith(sessionToken, trim(email), password, {
+              keys: false,
+              reason: SignInReasons.SIGN_IN,
+              redirectTo: REDIRECT_TO,
+              service: NON_SYNC_SERVICE
+            }));
+
+            // These should not be returned by default
+            assert.isFalse('unwrapBKey' in sessionData);
+            assert.isFalse('keyFetchToken' in sessionData);
+          });
+      });
+
+      it('passes along an optional `reason`', function () {
+        sinon.stub(relier, 'wantsKeys').callsFake(function () {
```

Comment:
These stubs are common on the optional params tests, worthwhile to pull into a describe block with `beforeEach`?

Reply:
`() => ` ?